### DISCUSSION TITLE
Feature request: External filtering plugins

### DISCUSSION THREAD

**Discussion by dov** [Motion-Project/motion#1506] [General]
Background. I have several cats at home, and they unfortunately set off a motion detection event of a camera that I have pointed to an external door.

So the idea is that it would be useful to connect motion to a classifier that can filter events of the pets.

More generally, motion should have an architecture in which it can send off preliminary motion events that it detected to an external classifier (plugin) that will give a go/no go, on whether to filter the event. In this way it would be possible to develop such classifiers externally with minimum changes to motion.

Ideas for such classifiers:

1. Use three images for extracting relative motions in the image. Then segment the moved areas and classify these as either human or other (pet, change of illumination, moving leaves).
2. Use single image classifiers, e.g. https://towardsdatascience.com/whats-new-in-yolov4-323364bb3ad3 .
3. Filter the event by the output of a PIR detector. 

I'm sure the community can come up with additional filters.

Note that this feature request is only about the changes to motion needed to support such external plugins.

**Comment by tosiara** [comment]
You can use `on_motion_detected` script to implement your logic and make a decision to start recording.
